tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -1018,20 +1009,19 @@ def _train_single_full(self, X_train, y_train, model: AbstractModel, X_val=None,
                     'num_cpus': self._scheduler_options['resource']['num_cpus'],
                     'num_gpus': self._scheduler_options['resource']['num_gpus'],
                 })  # Additional configurations for model.fit
-            model_names_trained = self._train_and_save(X_train, y_train, model, X_val, y_val, stack_name=stack_name, level=level, **model_fit_kwargs)
+            model_names_trained = self._train_and_save(X_train, y_train, model, X_val, y_val, X_unlabeled, stack_name=stack_name, level=level, **model_fit_kwargs)
         self.save()
         return model_names_trained
 
     # TODO: How to deal with models that fail during this? They have trained valid models before, but should we still use those models or remove the entire model? Currently we still use models.
     # TODO: Time allowance can be made better by only using time taken during final model training and not during HPO and feature pruning.
     # TODO: Time allowance not accurate if running from fit_continue
     # TODO: Remove level and stack_name arguments, can get them automatically
-    def _train_multi_repeats(self, X_train, y_train, X_val, y_val, models: list, k_fold, n_repeats, n_repeat_start=1, stack_name='core', level=0, time_limit=None, time_limit_total_level=None) -> List[str]:
+    def _train_multi_repeats(self, X_train, y_train, X_val, y_val, X_unlabeled, models: list, k_fold, n_repeats, n_repeat_start=1, stack_name='core', level=0, time_limit=None, time_limit_total_level=None) -> List[str]:
```

Comment:
later we should make sure that the pretraining on X_labeled only happens 1 time rather than during every fold of bagging. It may be easier to fix this after the API has been split into separate pretrain / finetune methods

Reply:
Added a TODO here.